backend/src/Tabulators/StarResults.js

```diff
@@ -1,759 +1,253 @@
+    nInvalidVotes: summaryData.nInvalidVotes,
+    nUnderVotes: summaryData.nUnderVotes,
+    nBulletVotes: summaryData.nBulletVotes,
+  }
+}
+
+function runStarRound(summaryData, remainingCandidates) {
+  const round_results = {
+    winners: [],
+    runner_up: [],
+    logs: [],
+  }
+  if (remainingCandidates.length === 1) {
+    round_results.winners.push(...remainingCandidates)
+    return round_results
+  }
+  const runoffCandidates = []
+  while (runoffCandidates.length < 2) {
+    const eligibleCandidates = remainingCandidates.filter(c => !runoffCandidates.includes(c))
+    const scoreWinners = getScoreWinners(summaryData, eligibleCandidates)
+    if (scoreWinners.length === 1) {
+      runoffCandidates.push(summaryData.candidates[scoreWinners[0].index])
+      round_results.logs.push(`${scoreWinners[0].name} advances to the runoff round with a score of ${summaryData.totalScores[scoreWinners[0].index].score}.`)
+    }
+    else {
+      round_results.logs.push(`${scoreWinners.map(c => summaryData.candidates[c.index].name).join(', ')} advances to score tiebreaker.`)
+      const scoreTieWinners = runScoreTiebreaker(summaryData, scoreWinners)
+      if (scoreTieWinners.length === 1) {
+        runoffCandidates.push(scoreTieWinners[0])
+        round_results.logs.push(`${scoreTieWinners[0].name} wins score tiebreaker and advances to the runoff round.`)
+      }
+      else {
+        if (scoreTieWinners.length === 2 && runoffCandidates.length === 0) {
+          runoffCandidates.push(...scoreTieWinners)
+          round_results.logs.push(`${scoreTieWinners[0].name} and ${scoreTieWinners[1].name} win score tiebreaker and advances to the runoff round.`)
         }
-      });
-
-      const totalScoreDelta =
-        candidates[a].totalScore - candidates[b].totalScore;
-      matrix[a][b] = {
-        result:
-          supportVotes > opposeVotes
-            ? "Win"
-            : supportVotes < opposeVotes
-            ? "Lose"
-            : totalScoreDelta > 0
-            ? "Win"
-            : totalScoreDelta < 0
-            ? "Lose"
-            : "TIE",
-        supportVotes: supportVotes,
-        support: support,
-        opposeVotes: opposeVotes,
-        oppose: oppose,
-        netVotes: netVotes,
-        netSupport: netSupport,
-        noPrefVotes: noPrefVotes
-      };
-
-      matrix[b][a] = {
-        result:
-          supportVotes < opposeVotes
-            ? "Win"
-            : supportVotes > opposeVotes
-            ? "Lose"
-            : totalScoreDelta < 0
-            ? "Win"
-            : totalScoreDelta > 0
-            ? "Lose"
-            : "TIE",
-        supportVotes: opposeVotes,
-        support: oppose,
-        opposeVotes: supportVotes,
-        oppose: support,
-        netVotes: -netVotes,
-        netSupport: -netSupport,
-        noPrefVotes: noPrefVotes
-      };
+        else {
+          const randomWinner = scoreTieWinners[getRandomInt(scoreTieWinners.length)]
```

Comment:
Not a code issue:  this seems like a huge problem.  Theoretically it is cheap to "run" the final round with either of the random winners.  Then we can present both outcomes  (if either runner up would lose, then simple.  but if either of them would potentially win the election???  would be good to surface that information clearly to the organizers)

From a code perspective:  if we are going to be using random for something important, we would want organizers to be able to pass in their own "random number" or whatever so they can have more trust here.  (or again, we could spit out the possible options as the result and let the organizers flip a coin or whatever)

Reply:
Seems like a pretty rare scenario, but we could spend a lot of time going down rabbit holes of various scenarios and Condorcet cycles. To keep the algorithm simple though we decided that ties in the score round should be decided just by who is tied and not considering who has already qualified for the runoff. We do want the tiebreaking protocol to be defined ahead of time but I suppose we could have a setting for if we just report the tie or resolve randomly (or use the 5 star tiebreaker that I completely forgot to add..). If we do resolve randomly it would be a good idea to have the seed be provided somehow. Maybe we can use the election ID as a seed or something. 

Also... I missed another part where if the number of tied candidates is equal to or less than the number of open seats, then they are all elected.